File: src/main/java/ru/ifmo/is/mfl/watchlists/dto/WatchListDto.java
package ru.ifmo.is.mfl.watchlists.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.users.dto.UserDto;

@Data
@EqualsAndHashCode(callSuper = true)
public class WatchListDto extends CrudDto {
  private int id;
  private UserDto user;
  private String name;
  private String photo;
  private boolean visibility;
  private int viewedCounter;
}

File: client/src/interfaces/watchlist/dto/WatchListDto.ts
import { CrudDto } from '@/interfaces/crud/dto/CrudDto';
import { UserDto } from '@/interfaces/user/dto/UserDto';

export interface WatchListDto extends CrudDto {
  user: UserDto;
  name: string;
  photo?: string;
  visibility: boolean;
  viewedCounter: number;
}
